### PR TITLE
Lcg/fix provider resolver api break

### PR DESCRIPTION
**Pull request by lamont-granquist**

### CODE REVIEW

**Comment by lamont-granquist** [comment]
@thommay @coderanger @chef/client-engineers @chef/client-core review

File: lib/chef/provider_resolver.rb [hunk 1]
```diff
@@ -47,7 +47,9 @@ def resolve
     def enabled_handlers
       @enabled_handlers ||=
         providers.select do |klass|
-          klass.provides?(node, resource.resource_name)
+          # NB: this is different from resoruce_resolver which must pass a resource_name
```
Comment (danielsdeleo) [inline lib/chef/provider_resolver.rb]: typo :space_invader:

**Comment by thommay** [comment]
Looks :+1: to me besides that typo

**Comment by btm** [comment]
:+1:

**Comment by tyler-ball** [comment]
:+1: 

Was there a bug related to this?

**Comment by btm** [comment]
@tyler-ball I was wondering if there was an issue filed too. I presume @lamont-granquist found this in his own testing?

The commit message:
```
Fix API break in 12.3.0.rc.0
provides? on a provider was passed the full resource.  provides? on
a resource class cannot be passed the resource and must be passed the
declared_type instead.  i tried to make them the same, but that would
change the API on the provider.
```

**Comment by thommay** [comment]
@coderanger mentioned it in IRC

**Comment by lamont-granquist** [comment]
lol typo fixed.

**Pull request merged**